geomesa-memory/src/main/scala/org/locationtech/geomesa/memory/cqengine/utils/CQEngineQueryVisitor.scala

```diff
@@ -0,0 +1,365 @@
+        val attr = lookup.lookupComparable[Date](prop.name)
+        val value = prop.literal.evaluate(null, classOf[Date])
+        new cqquery.simple.LessThan[SimpleFeature, Date](attr, value, false)
+      }
+      case c => throw new RuntimeException(s"Before: $c not supported")
+    }
+  }
+
+  /**
+    * During: only for time attributes, and is exclusive at both ends
+    */
+  override def visit(during: During, extraData: scala.Any): AnyRef = {
+    val prop = getProp(during)
+    sft.getDescriptor(prop.name).getType.getBinding match {
+      case c if classOf[Date].isAssignableFrom(c) => {
+        val attr = lookup.lookupComparable[Date](prop.name)
+        val p = prop.literal.evaluate(null, classOf[Period])
+        val lower = p.getBeginning.getPosition.getDate
+        val upper = p.getEnding.getPosition.getDate
+        new cqquery.simple.Between[SimpleFeature, java.util.Date](attr, lower, false, upper, false)
+      }
+      case c => throw new RuntimeException(s"During: $c not supported")
+    }
+  }
+
+  /**
+    * BinaryTemporalOperator: Fallback non-indexable implementation of other temporal operators.
+    * Handles:
+    *     AnyInteracts, Begins, BegunBy, EndedBy, Ends, Meets,
+    *     MetBy, OverlappedBy, TContains, TEquals, TOverlaps
+    */
+  override def visit(filter: BinaryTemporalOperator, data: scala.Any): AnyRef = handleGeneralCQLFilter(filter)
+
+  def handleGeneralCQLFilter(filter: Filter): Query[SimpleFeature] = {
+    new GeoToolsFilterQuery(filter)
+  }
+
+  /**
+    * Build a PropertyLiteral for every expression with a property name in it
+    * (essentially a wrapper around getAttributeProperty)
+    */
+  def getProp(filter: Filter): PropertyLiteral = {
+    val prop = filter match {
+      case f: BinarySpatialOperator => checkOrder(f.getExpression1, f.getExpression2)
+      case f: BinaryTemporalOperator => checkOrder(f.getExpression1, f.getExpression2)
+      case f: PropertyIsNotEqualTo => checkOrder(f.getExpression1, f.getExpression2)
+      // we support a wider range of PropertyIsLike filters than getAttributeProperty does
+      case f: PropertyIsLike => {
+        val propName = f.getExpression.asInstanceOf[PropertyName].getPropertyName
+        Some(PropertyLiteral(propName, ff.literal(f.getLiteral), None))
+      }
+      case f: PropertyIsNull => {
+        val propName = f.getExpression.asInstanceOf[PropertyName].getPropertyName
+        Some(PropertyLiteral(propName, ff.literal(null), None))
+      }
+      case f: PropertyIsNil => {
+        val propName = f.getExpression.asInstanceOf[PropertyName].getPropertyName
+        Some(PropertyLiteral(propName, ff.literal(null), None))
+      }
+      case f => getAttributeProperty(f)
+    }
+    prop match {
+      case Some(p) => p
+      case None => throw new RuntimeException(s"Can't parse filter $filter")
+    }
+  }
+
+  // TODO: revisit this if needed.
+  override def toString = "CQEngineQueryVisit()"
```

Comment:
Yeah... let's try and ditch this.  Maybe I implemented it due to something in the REPL?